File: benchmarks/fib/fib.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/time.h>

long fib(int n) {
    if (n <= 1) return n;
    return fib(n - 1) + fib(n - 2);
}

int main(int argc, char *argv[]) {
    int n = argc > 1 ? atoi(argv[1]) : 35;

    struct timeval start, end;
    gettimeofday(&start, NULL);

    long result = fib(n);

    gettimeofday(&end, NULL);
    double elapsed_ms = (end.tv_sec - start.tv_sec) * 1000.0 + (end.tv_usec - start.tv_usec) / 1000.0;
    fprintf(stderr, "TIME_MS:%.2f\n", elapsed_ms);

    printf("%ld\n", result);
    return 0;
}

File: benchmarks/fib/fib.py
import sys
import time

def fib(n):
    if n <= 1:
        return n
    return fib(n - 1) + fib(n - 2)

n = int(sys.argv[1]) if len(sys.argv) > 1 else 35

start = time.perf_counter()
result = fib(n)
elapsed = (time.perf_counter() - start) * 1000
print(f"TIME_MS:{elapsed:.2f}", file=sys.stderr)
print(result)
